src/download.rs

```diff
@@ -70,21 +78,34 @@ impl DownloadCfg {
 
     pub fn from_package(name: &str, package: Package) -> Result<Self> {
         let target = TargetTriple::from_component(name)?;
-        let tarball_name = tarball_name(name, &package.version, &target)?;
+        let tarball_name =
+            tarball_name(name, &package.version.semver, package.version.date, &target)?;
         let tarball_url = package.target[&target.to_string()].url.clone();
         let hash = Some(package.target[&target.to_string()].hash.clone());
         Ok(Self {
             name: name.to_string(),
             target,
-            version: package.version,
+            date: package.version.date,
+            version: package.version.semver,
             tarball_name,
             tarball_url,
             hash,
         })
     }
 }
 
-pub fn tarball_name(name: &str, version: &Version, target: &TargetTriple) -> Result<String> {
+pub fn tarball_name(
+    name: &str,
+    version: &Version,
+    date: Option<Date>,
+    target: &TargetTriple,
+) -> Result<String> {
+    let version = if let Some(d) = date {
+        version.to_string() + "-" + &d.to_string()
+    } else {
+        version.to_string()
+    };
```

Comment:
Could we potentially be using a `Display` impl for `PackageVersion` here instead? E.g.
```rust
let version_str = format!("{}", version);
```

Reply:
Hmm, the tarball names for nightlies actually don't include the semver [(link)](https://github.com/FuelLabs/sway-nightly-binaries/releases), eg. `forc-binaries-nightly-<date>-<target>.tar.gz` (where `date` is in the format `YYYY-MM-DD`) . So a `Display` impl wouldn't work here since we would just want `nightly-<date>`, and we'd have to handle this case differently.

Do we want to include the semver in the releases for nightlies? We can scrub the existing releases (since no one is using them right now anyway) and re-release with semvers attached.

From what I saw when searching for standards on releasing nightlies it seems like some choose to omit the semver.

This is an issue also with `forc-binaries` because the releases don't include the semver, but that's unrelated.

Reply:
> Hmm, the tarball names for nightlies actually don't include the semver

Ahh In that case feel free to ignore this suggestion!

Reply:
>Do we want to include the semver in the releases for nightlies?

I still think this is a something to consider though I'm not sure if this matters. I guess it would be nice if we include the semver in the tarball names? But it wouldn't be a strictly necessary thing.

Reply:
I'm happy to go either way on this one :+1: Any precedent set by Rust we can follow?

Reply:
Just for posterity, it seems like from browsing other projects that there's no clear standard to doing this. I decided to re-release the binaries with the semver included, as well as re-publish the channels to include the updated URLs.

IMO, having the semver included in the releases as well as the URLs could clear up confusion about which latest release a certain nightly build was last tagged to; and we at least have those version numbers to refer to when working with nightlies. It's also a piece of info that wouldn't cost much to include (slightly longer tarball name), so why not.

Will be nuking the old releases/channels that don't conform to this standard since nightlies aren't live yet.